src/Rings/mpoly-graded.jl

```diff
@@ -978,14 +978,20 @@ true
 function is_homogeneous(F::MPolyDecRingElem)
   D = parent(F).D
   d = parent(F).d
-  S = Set{elem_type(D)}()
+  S = nothing
+  u = zero(D)
   for c = MPolyExponentVectors(forget_decoration(F))
-    u = parent(F).D[0]
+    zero!(u.coeff)
+    # TODO: once Hecke supports zero! on FinGenAbGroupElem, switch to this:
+    #zero!(u)
     for i=1:length(c)
-      u += c[i]*d[i]
+      addmul!(u.coeff, d[i].coeff, c[i])
+      # TODO: once Hecke supports addmul! on FinGenAbGroupElem, switch to this:
+      #addmul!(u, d[i], c[i])
```

Comment:
This is not correct:
```
julia> G = abelian_group([2, 2])
(Z/2)^2

julia> u = G([0, 0])
Abelian group element [0, 0]

julia> d = G([0, 1])
Abelian group element [0, 1]

julia> addmul!(u.coeff, d.coeff, ZZ(2))
[0   2]

julia> u
Abelian group element [0, 2]
```
In `G` there shouldn't be an element with coefficients `[0, 2]`.

Reply:
Indeed, a final reduction is necessary. See also my Hecke PR adding a helper for that.

Reply:
Assuming https://github.com/thofma/Hecke.jl/pull/1619 is merged and released soonish, can switch to that, too...

Reply:
This is now resolved. I've also updated the measurements in the PR description (we allocate a tad more and things got slightly slower again, but it's still a clear win)